src/test/groovy/streama/VideoServiceSpec.groovy

```diff
@@ -0,0 +1,47 @@
+package streama
+
+import grails.test.mixin.TestFor
+import spock.lang.Specification
+
+@TestFor(VideoService)
+class VideoServiceSpec extends Specification {
+
+  def fileServiceMock = Mock(FileService)
+
+  def setup() {
+   service.fileService = fileServiceMock
+  }
+
+  void "test delete video and all associations"() {
+    setup:
+      GroovyMock(ViewingStatus, global: true)
+
+      def viewingStatus1 = Mock(ViewingStatus)
+      def viewingStatus2 = Mock(ViewingStatus)
+
+      def file1 = Mock(File)
+      def file2 = Mock(File)
+
+      Video video = Mock(Video)
+      video.files >> [file1, file2]
+
+      ViewingStatus.findAllByVideo(video) >> [viewingStatus1, viewingStatus2]
+
+    when: 'Method deleteVideoAndAssociations is called'
+      service.deleteVideoAndAssociations(video)
+
+    then: 'The video has been flagged to delete'
+      1 * video.setDeleted(true)
```

Comment:
Hello @dularion, this is assertion syntax in mock objects, in this form, i expected the method setDeleted will be called one time with param true.

If you want more information: http://spockframework.org/spock/docs/1.1-rc-3/interaction_based_testing.html